src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -4346,10 +4346,10 @@ FIXTURE_TEST(test_offset_range_size_compacted, storage_test_fixture) {
 
 FIXTURE_TEST(test_offset_range_size2_compacted, storage_test_fixture) {
 #ifdef NDEBUG
-    size_t num_test_cases = 5000;
+    size_t num_test_cases = 1000;
```

Comment:
why reduce the non-debug mode too?

Reply:
IIRC, I've seen this fixture test time out in both modes- it's been flakey for a few months. https://github.com/redpanda-data/redpanda/issues/18563